### DISCUSSION TITLE
Relative path for InverseSearchCmdLine?

### DISCUSSION THREAD

**Discussion by kromuchi** [sumatrapdfreader/sumatrapdf#3340] [Q&A]
In the SumatraPDF Advanced Settings, it is possible to specify the path / command line for an inverse search, e.g.

`InverseSearchCmdLine = "C:\Program Files (x86)\TeXnicCenter\TeXnicCenter.exe" /nosplash /ddecmd "[goto('%f', '%l')]"`

Is it also possible to specify an **relative path** (e.g. to call a portable app from a portable sumtrapdf? (e.g. like `InverseSearchCmdLine = "..\..\TeXnicCenter\TeXnicCenter.exe" /nosplash /ddecmd "[goto('%f', '%l')]"`)
And if so, which is the current directory (the directory where sumatrapdf.exe is located or where the document is placed - btw. I tried both versions and failed but I don't know if I made a mistake...)?

Thank you very much.

**Comment by kjk** [comment]
The comment in source code says:
```
        // resolve relative paths with relation to SumatraPDF.exe's directory
```

Does it actually work? I don't think I have tested it.

Try the simplest: put your `TeXnicCenter.exe` next to `SumatraPDF.exe` and try `TeXnicCenter.exe` and `.\TeXnicCenter.exe`.

**Reply by kromuchi** [reply]
Thank you for the fast reply. I did a few tests:

If `SumatraPDF.exe` and `TeXnicCenter.exe` are in the same directory, this line works:
`InverseSearchCmdLine = "TeXnicCenter.exe" /nosplash /ddecmd "[goto('%f', '%l')]"`

However, these lines don't work:
`InverseSearchCmdLine = ".\TeXnicCenter.exe" /nosplash /ddecmd "[goto('%f', '%l')]"`
`InverseSearchCmdLine = "./TeXnicCenter.exe" /nosplash /ddecmd "[goto('%f', '%l')]"`

If `SumatraPDF.exe` is one level above `TeXnicCenter.exe` both lines don't work:
`InverseSearchCmdLine = "..\TeXnicCenter.exe" /nosplash /ddecmd "[goto('%f', '%l')]"`
`InverseSearchCmdLine = "../TeXnicCenter.exe" /nosplash /ddecmd "[goto('%f', '%l')]"`

So for me, I have a workaround (they have to be in the same folder), but maybe it's worth fixing (on the other hand it seems I'm the first to have asked for this feature in over 15 years, so ... ;)

PS. I tested it with 3.6.4 64-bit.

**Reply by GitHubRulesOK** [reply]
>If SumatraPDF.exe is one level above TeXnicCenter.exe both lines don't work:

means TeXnicCenter.exe needs ./TeXnicCenter/TeXnicCenter.exe as its below SumatraPDF

**Reply by kromuchi** [reply]
> > If SumatraPDF.exe is one level above TeXnicCenter.exe both lines don't work:
> 
> means TeXnicCenter.exe needs ./TeXnicCenter/TeXnicCenter.exe as its below SumatraPDF

totally true, my fault. `./TeXnicCenter/TeXnicCenter.exe` doesn't work but `TeXnicCenter/TeXnicCenter.exe` works.

But when actually `SumatraPDF.exe` is in a subfolder of `TeXnicCenter.exe` or there is another relative path required to go at least one level up, neither `../TeXnicCenter.exe` nor `..\TeXnicCenter.exe` (or in the other example `../TeXnicCenter/TeXnicCenter.exe` nor `..\TeXnicCenter\TeXnicCenter.exe`) works.

**Reply by GitHubRulesOK** [reply]
Hmm I did have a mistake/error message saying wrong path when trying relative as I added too many ..\ however for me this is working

sumatrapdf.exe and the data files are in one folder so I wish the files to open in the editor that is 2 folders up and one down thus 
options `..\..\PDFXEdit9_Portable\PDFXEdit.exe "%f" %l` works (EXCEPT that Xchange grumbles wrong data as it probably should in this case)

![image](https://user-images.githubusercontent.com/12754913/224182154-0572a447-cf0a-487f-b04b-2948c5ffe671.png)

If I move the data down a folder then it fails as the PDF data folder is no longer relative to the PDF editor.


**So the editor inverse call is relative to the data directory.**

**Reply by GitHubRulesOK** [reply]
@kromuchi note that last comment that for the pdf and companion files the transfer is in a relative case passed from data to editor thus its the project directory relative to the editor folder that counts.

**Comment by GitHubRulesOK** [comment]
hmm current working directory when using portable apps can vary by calling method, but should not matter as its absolute data filenames via tokens so SumatraPDF will be called with the folder path/file.PDF and will look in that pdf directory for the .synctex file which should by tokens call that same folder with the .tex filename HOWEVER on occasion the synctex file may have a linux mount alias for the windows drive: device which then could cause issues. Ideally all files should be built at exactly the same folder location.

the tokenisation of filenames in TexnicCentre is called "placeholders" and so much broader than many other TeX IDEs they have their own manual https://github.com/GitHubRulesOK/MyNotes/raw/master/AppNotes/SumatraPDF/SumatraPDF%20with%20TeXnicCenter.pdf

>If you have multiple (or portable) editors you can set up BATch ComManDs to change the editor name/path 
via an external -inverse-search call to SumatraPDF however it only needs to be done once per session 
not at every internal call from the editor.
